graql/executor/property/HasAttributeExecutor.java

```diff
@@ -135,20 +134,24 @@ public VarProperty property() {
 
         @Override
         public Set<Variable> requiredVars() {
-            Set<Variable> required = new HashSet<>();
-            required.add(var);
-            required.add(property.attribute().var());
-            return Collections.unmodifiableSet(required);
+            return Collections.unmodifiableSet(vars());
         }
 
         @Override
         public Set<Variable> producedVars() {
-            return ImmutableSet.of();
+            return Collections.emptySet();
+        }
+
+        private Set<Variable> vars() {
+            Set<Variable> vars = new HashSet<>();
+            vars.add(var);
+            vars.add(property.attribute().var());
+            return vars;
         }
 
         @Override
         public TiebreakDeletionOrdering ordering(WriteExecutor executor) {
-            return TiebreakDeletionOrdering.EDGE;
+            return TiebreakDeletionOrdering.HAS;
         }
```

Comment:
This method takes in a `WriteExecutor` but not use it. Does the other implementation of the interface where this method comes from uses it properly? Some feels off...